schema/src/main/scala/org/constellation/schema/signature/Signable.scala

```diff
@@ -8,15 +8,21 @@ trait Signable {
 
   protected def toEncode: AnyRef = this
 
+  /**
+    * Override this value with false if data representing this class is also serialized and deserialized in format other
+    * than Kryo (i.e. JSON for transmission over the network).
+    */
```

Comment:
Can you give more detail here, because I think it's not strictly connected to whether it's serialized/deserialized to e.g. json or not, right?

Reply:
Yes it is related JSON. 

References in the graph of objects are not reflected in JSON. Example:

Given this list:
```
val x = new String("x")
val list1 = List(x, x)
```
if we serialize it as sender to JSON
[ "x", "x"]
And deserialize it as receiver from JSON into:
```
val list2 = List(new String("x"), new String("x")
```
Then with default settings, the hash of the list calculated on the sender side will be different from the hash calculated on the receiver side, even though both `list1` and `list2` are semantically the same.

Reply:
I kind of get it, but we are also building other structures out of data send through endpoints and when we receive that data it gets their own references locally, and then we are e.g. building a Seq[String] hashing it with Kryo locally and the hashes across different nodes match. Why is that bit working?

Reply:
E.g. snapshot hash is madef out of Seq[Strings] where these strings are hashes of CheckpointBlock, and we are sending CheckpointBlocks over an endpoint if I'm not mistaken.
I'm just wondering, I'm not saying you're not right :).

Reply:
I was wandering the same. One explanation could be that the graph of objects that is serialized and deserialized contains unique values (graph then becomes a tree essentially). From your experience, do you know any examples where this could not be the case (the graph is not a tree)?

Reply:
But I wonder even if it holds unique values, then still these values have a different reference locally. If the difference between a tree and a graph would count it would mean that Kryo ignores references when it encounters tree but doesn't ignore when it encounters a graph or something like that, right?

Reply:
I think I don't understand the question. Let's talk offline for clarification.